File: frontend/edenBooks/src/modules/reviews/services/reviewService.ts
import { reviewApi } from "../api/reviewApi";
import { IReview } from "../interfaces/IReview";
import { IReviewService } from "../interfaces/IReviewService";

export const ReviewService: IReviewService = {
  GetReviews: async () => reviewApi.GetReviews(),
  GetReviewById: async (id: number) => reviewApi.GetReviewById(id),
  UpdateReview: async (id: number, review: IReview) =>reviewApi.UpdateReview(id, review),
  DeleteReview: async (id: number) => reviewApi.DeleteReview(id),
};

File: backend/internal/modules/reviews/domain/ports/services/review_service.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/reviews/domain/entities"
)

type ReviewService interface {
	CreateReview(review *entities.Review) error

	UpdateReview(r *entities.Review) error

	DeleteReview(id uint) error

	GetReviewByID(id uint) (*entities.Review, error)

	GetFilteredReviews(filters map[string]string) ([]entities.Review, error)
}
